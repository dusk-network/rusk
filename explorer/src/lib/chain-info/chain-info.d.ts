type BlockHeader = {
  date: Date;
  feespaid: number;
  hash: string;
  height: number;
  nextblockhash: string;
  prevblockhash: string;
  reward: number;
  seed: string;
  statehash: string;
};

type BlockTransactions = {
  data: Transaction[];
  stats: {
    averageGasPrice: number;
    gasLimit: number;
    gasUsed: number;
  };
};

type Block = {
  header: BlockHeader;
  transactions: BlockTransactions;
};

type ChainInfo = {
  blocks: Block[];
  transactions: Transaction[];
};

type SearchResult = {
  id: string;
  type: "account" | "block" | "transaction";
};

type Transaction = {
  amount: number | undefined;
  blobHashes: string[];
  from: string | undefined;
  to: string | undefined;
  blockhash: string;
  blockheight: number;
  date: Date;
  feepaid: number;
  gaslimit: number;
  gasprice: number;
  gasspent: number;
  memo: string;
  method: string;
  success: boolean;
  txerror: string;
  txid: string;
  txtype: string;
  payload: string;
  nonce: number | undefined;
};
