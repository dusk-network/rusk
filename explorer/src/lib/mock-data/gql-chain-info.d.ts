type GQLBlock = {
  fees: number;
  gasSpent: number;
  header: {
    gasLimit: number;
    hash: string;
    height: number;
    nextBlockHash?: string;
    prevBlockHash: string;
    seed: string;
    stateHash: string;
    timestamp: number;
  };
  reward: number;
  transactions: GQLTransaction[];
};

type GQLCallData = {
  contractId: string;
  fnName: string;
};

type GQLChainInfo = {
  blocks: GQLBlock[];
  transactions: GQLTransaction[];
};

type GQLSearchResult = {
  block?: {
    header: {
      hash: string;
    };
  } | null;
  tx?: {
    id: string;
  } | null;
  account?: {
    id: string;
  } | null;
};

type GQLTransaction = {
  blockHash: string;
  blockHeight: number;
  blockTimestamp: number;
  err: string | null;
  gasSpent: number;
  id: string;
  tx: {
    blobHashes: string[] | null;
    callData: GQLCallData | null;
    gasLimit: number;
    gasPrice: number;
    id: string;
    isDeploy: boolean;
    memo: string;
    txType: string;
    json?: string;
  };
};
