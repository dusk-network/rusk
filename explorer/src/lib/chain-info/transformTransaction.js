import { unixTsToDate } from "$lib/dusk/date";

/** @type {(v: GQLTransaction) => Transaction} */
const transformTransaction = (tx) => ({
  blockhash: tx.blockHash,
  blockheight: tx.blockHeight,
  date: unixTsToDate(tx.blockTimestamp),
  feepaid: tx.gasSpent * tx.tx.gasPrice,
  from: undefined,
  gaslimit: tx.tx.gasLimit,
  gasprice: tx.tx.gasPrice,
  gasspent: tx.gasSpent,
  memo: tx.tx.memo ?? "",
  method: tx.tx.isDeploy ? "deploy" : (tx.tx.callData?.fnName ?? "transfer"),
  success: tx.err === null,
  to: undefined,
  txerror: tx.err ?? "",
  txid: tx.id,
  txtype: tx.tx.txType,
});

export default transformTransaction;
