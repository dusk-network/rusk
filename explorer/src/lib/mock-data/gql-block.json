{
  "block": {
    "fees": 580718,
    "gasSpent": 580718,
    "header": {
      "gasLimit": 5000000000,
      "hash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
      "height": 495868,
      "nextBlockHash": "6011556208a85e6001bd01ccbf936486b91318a7f6cbcf7ab810adf6fae34204",
      "prevBlockHash": "07b74b35c2c7cf8f41426cd0870bafa1a2c7adee3fdd876643548096186fc4cb",
      "seed": "af15447e3a004a79d4ae8b084f7b76b78d95880bb63e1cfa79250a310731f52e6d84ee42a5d6fc2cb99c5b1f489761f6",
      "stateHash": "20bb0a677b93f084afadfd34bec3ac3feee33a020b81d9549afa2268e8543acb",
      "timestamp": 1713298022
    },
    "reward": 16000000000,
    "transactions": [
      {
        "blockHash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
        "blockHeight": 495868,
        "blockTimestamp": 1713298022,
        "err": null,
        "gasSpent": 290866,
        "id": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
        "tx": {
          "blobHashes": null,
          "callData": null,
          "isDeploy": false,
          "gasLimit": 500000000,
          "gasPrice": 1,
          "id": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
          "memo": "",
          "txType": "Moonlight"
        }
      },
      {
        "blockHash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
        "blockHeight": 495868,
        "blockTimestamp": 1713298022,
        "err": null,
        "gasSpent": 289852,
        "id": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
        "tx": {
          "blobHashes": null,
          "callData": null,
          "gasLimit": 500000000,
          "gasPrice": 1,
          "id": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
          "isDeploy": false,
          "memo": "some notes about the transaction",
          "txType": "Phoenix"
        }
      }
    ]
  }
}
