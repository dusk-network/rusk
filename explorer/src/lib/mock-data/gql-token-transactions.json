[
  {
    "amount": 4359,
    "from": "0xf1c4230aca3ca19199d0f9665bbd9cff8",
    "to": "0x189f8d571b05adc6a35ef1bb0e3b458184",
    "blockhash": "103749d1e8990cdda301b2c4efd6c50735548dd2fff511d9d99c75",
    "blockheight": 6670,
    "date": "2025-02-04 21:15:22.373891+00:00",
    "feepaid": 364,
    "gaslimit": 40133828,
    "gasprice": 21,
    "gasspent": 8255387,
    "memo": "Mock transaction",
    "method": "stake",
    "success": true,
    "txerror": "Out of gas",
    "txid": "15fb0001ed66bb0dacc02ec003de09e31a3de10e810fa8bb384175",
    "txtype": "Phoenix",
    "payload": "0x",
    "nonce": 0
  }
]
