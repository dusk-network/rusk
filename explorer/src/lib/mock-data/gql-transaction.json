{
  "tx": {
    "blockHash": "3c6e4018cfa86723e50644e33d3990bc27fc794f6b49fbf6290e4d308e07bd2d",
    "blockHeight": 487166,
    "blockTimestamp": 1713249549,
    "gasSpent": 290766,
    "id": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
    "err": null,
    "tx": {
      "blobHashes": null,
      "callData": null,
      "gasLimit": 500000000,
      "gasPrice": 1,
      "id": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
      "isDeploy": false,
      "memo": "some notes about the transaction",
      "txType": "Moonlight",
      "json": "{\"type\":\"moonlight\",\"sender\":\"kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj\",\"receiver\":\"mHWZo9qYUhqp2SEmtWN7EuDKFwVrjgdEyzpHdfLB6R9scRAq5EkUQyLB9fBfCGt1wjxfEpFxHq9MBPVVPY3Lk3JKnQLWZzj7UYAR4mGmeGQZwCaeCS8uA63ZPKnGpjiUnj\",\"value\":9812378912731,\"nonce\":156,\"deposit\":0,\"fee\":{\"refund_address\":\"kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj\",\"gas_limit\":\"2500000\",\"gas_price\":\"1\"},\"call\":null,\"is_deploy\":false,\"memo\":null}"
    }
  }
}
