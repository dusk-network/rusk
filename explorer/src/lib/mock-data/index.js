export { default as apiMarketData } from "./api-market-data.json";
export { default as apiNodeInfo } from "./api-node-info.json";
export { default as apiNodeLocations } from "./api-node-locations.json";
export { default as apiStats } from "./api-stats.json";
export { default as gqlBlock } from "./gql-block.json";
export { default as gqlBlockDetails } from "./gql-block-details.json";
export { default as gqlBlocks } from "./gql-blocks.json";
export { default as gqlLatestChainInfo } from "./gql-latest-chain-info.json";
export { default as gqlSearchPossibleResults } from "./gql-search-possible-results.json";
export { default as gqlTransaction } from "./gql-transaction.json";
export { default as gqlTransactions } from "./gql-transactions.json";
export { default as hostProvisioners } from "./host-provisioners.json";
export { default as nodeLocationsCount } from "./node-locations.count.json";

export { default as accounts } from "./accounts.json";
export { default as tokens } from "./tokens.json";
export { default as gqlTokenTransactions } from "./gql-token-transactions.json";
