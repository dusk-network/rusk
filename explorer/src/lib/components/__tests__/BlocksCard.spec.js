import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { compose, mapWith, take } from "lamb";

import { gqlBlocks } from "$lib/mock-data";
import { transformBlock } from "$lib/chain-info";
import { appStore } from "$lib/stores";
import { BlocksCard } from "..";

describe("Blocks Card", () => {
  vi.useFakeTimers();
  vi.setSystemTime(new Date(2024, 4, 30));

  const getTenBlocks = compose(mapWith(transformBlock), take(10));
  const data = getTenBlocks(gqlBlocks.blocks);

  const baseProps = {
    blocks: null,
    error: null,
    loading: false,
    appStore: appStore
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  afterAll(() => {
    vi.useRealTimers();
  });

  it("should render the `BlocksCard` component", () => {
    const { container } = render(BlocksCard, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the `Show More` button is the card is in the loading state", () => {
    const loading = true;

    const { container, getByRole } = render(BlocksCard, {
      ...baseOptions,
      props: { ...baseProps, loading },
    });

    expect(getByRole("button")).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the `Show More` button if there is no more data to display", async () => {
    const loading = false;
    const blocks = data;

    const { container, getByRole } = render(BlocksCard, {
      ...baseOptions,
      props: { ...baseProps, blocks, loading },
    });

    expect(getByRole("button")).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });
});
