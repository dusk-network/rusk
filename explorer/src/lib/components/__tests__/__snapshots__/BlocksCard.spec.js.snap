// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Blocks Card > should disable the \`Show More\` button if there is no more data to display 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Blocks â€” 10 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      disabled=""
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  <div
    class="data-card__content"
  >
    <div
      class="table-container blocks-table data-card__table"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="table__row"
          >
            <th
              class="table__header-cell align-left"
            >
              Block
            </th>
            
             
            <th
              class="table__header-cell align-left"
            >
              Gas
            </th>
            
             
            <th
              class="table__header-cell align-left"
            >
              Txn(s)
            </th>
            
             
            <th
              class="table__header-cell align-left"
            >
              Rewards (Dusk)
            </th>
            
          </tr>
        </thead>
         
        <tbody
          class="table__body"
        >
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=3d43be31a8cb5d3f61b4e8391800a01033d650b11c4c874210dbaa3c118474a6"
              >
                1,365,454
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:54.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=2bc290511be774ec9c04e6c0ceb2469798242b997422c542a7d6b0b6f3368414"
              >
                1,365,453
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:51.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=02f38d2e2a986cf77a70e8fe4f333ef9c5e075987891f3f526c8a57590800178"
              >
                1,365,452
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:47.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=c79e31ce8e6133ebfb0d85649a12167769ddc83420550faebc536e613d74d676"
              >
                1,365,451
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:45.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=22ffe1b159e1b33b06d566b3ce3119af6f4bec4d2a3f6916560fe3cb30fcc1ef"
              >
                1,365,450
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:42.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=79c2d3213df80f8113d1ca16c49b1818fb5c31d1437e0a447719491a9c1b5f85"
              >
                1,365,449
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:39.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=fb3e82e43b141409156f935afc69ccb09a59a72a10258401b539123981db571c"
              >
                1,365,448
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:35.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=4b402a917bf42561096a9224e6451b77b8cbdc50272013442eccb5fcc542ef12"
              >
                1,365,447
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:32.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=38d069620a7cbad7cadb10ecc797b1344284b6324ac85f5e126226a08d47398a"
              >
                1,365,446
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:29.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface block__link"
                href="/some-base-path/blocks/block?id=14c7d9f5e44d599a167d82a449d73daf54e32bf002c6cd5893de5f904024922a"
              >
                1,365,445
              </a>
               
              <time
                class="dusk-relative-time block__time"
                datetime="2024-05-28T14:26:24.000Z"
              >
                yesterday
              </time>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="block__fee-avg-label"
              >
                AVG PRICE:
              </b>
               
              0
              <br />
               
              <b
                class="block__fee-total-label"
              >
                USED:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              16
            </td>
            
             
          </tr>
          
        </tbody>
      </table>
    </div>
    
  </div>
  
   
</div>
`;

exports[`Blocks Card > should disable the \`Show More\` button is the card is in the loading state 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Blocks â€” 0 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      disabled=""
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  <div
    aria-label="Loading"
    class="dusk-progress-bar data-card__progress-bar"
    role="progressbar"
  >
    <div
      class="dusk-progress-bar__filler dusk-progress-bar__filler--undetermined"
    />
  </div>
  
   
</div>
`;

exports[`Blocks Card > should hide the \`Show More\` button if an error has occurred 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Blocks â€” 0 Displayed Items
    </h1>
     
  </header>
   
  <div
    class="alert"
  >
    <svg
      class="dusk-icon dusk-icon--size--large"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,2L1,21H23M12,6L19.53,19H4.47M11,10V14H13V10M11,16V18H13V16"
      />
    </svg>
    
     
    <header>
      <p>
        There was an error fetching the data.
      </p>
       
      <p
        class="alert__error-message"
      >
        error
      </p>
    </header>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--secondary dusk-button--size--normal"
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Retry
      </span>
    </button>
  </div>
  
   
</div>
`;

exports[`Blocks Card > should render the \`BlocksCard\` component 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Blocks â€” 0 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  
   
</div>
`;
